src/services/CertificateGenerationService.ts

```diff
@@ -194,34 +202,85 @@ class CertificateGenerationService {
    * Check if the specific test station is in Wales
    * @param testStations list of all test stations
    * @param testStationPNumber pNumber from the test result
-   * @returns boolean
+   * @returns string postcode of the pNumber test station
    */
-  public isTestStationWelsh(testStations: ITestStation[], testStationPNumber: string) {
-    // default parameter value so that if test station cannot be determined, processing will continue
-    let isWelsh = false;
-
+  public getThisTestStation(
+    testStations: ITestStation[],
+    testStationPNumber: string
+  ) {
     if (!testStations || testStations.length === 0) {
       console.log(`Test stations data is empty`);
-      return isWelsh;
+      return null;
     }
 
     // find the specific test station by the PNumber used on the test result
-    const pNumberTestStation = testStations.filter((x) => {
+    const thisTestStation = testStations.filter((x) => {
       return x.testStationPNumber === testStationPNumber;
     });
 
-    if ((pNumberTestStation) && (pNumberTestStation.length > 0)) {
-      const thisTestStation = pNumberTestStation[0];
-      if ((thisTestStation.testStationCountry) && (thisTestStation.testStationCountry.toUpperCase() === ATF_COUNTRIES.WALES)) {
-        isWelsh = true;
-      }
-      console.log(`Test station details: ${thisTestStation.testStationPNumber} ${thisTestStation.testStationName} in ${thisTestStation.testStationCountry}`);
+    if (thisTestStation && thisTestStation.length > 0) {
+      return thisTestStation[0].testStationPostcode;
     } else {
-      console.log(`Test station details could not be found for ${testStationPNumber} `);
+      console.log(
+        `Test station details could not be found for ${testStationPNumber}`
+      );
+      return null;
     }
+  }
 
-    console.log(`Return value for isWelsh is ${isWelsh}`);
-    return isWelsh;
+  /** Call SMC postcode lookup with test station postcode
+   * @param postcode
+   * @returns boolean true if Welsh
+   */
+  public async lookupPostcode(postcode: string) {
+    const secretConfig = await this.getSecret();
+
+    if (secretConfig) {
+      const addressResponse: boolean = await axios({
+        method: "get",
+        url: secretConfig.url + "/" + postcode,
+        headers: {"x-api-key": secretConfig.key},
+      })
+          .then((response) => {
+            return response.data.isWelshAddress;
+          })
+          .catch((error) => {
+            console.log(`Error looking up postcode ${postcode}`);
+            console.log(error);
+            return false;
+          });
+
+      console.log(`Return value for isWelsh for ${postcode} is ${addressResponse}`);
```

Comment:
I don't think we want to be logging out postcodes, can this log be removed

Reply:
It's there to aid support and testing